modules/sdk-core/src/account-lib/mpc/hdTree.ts

```diff
@@ -111,8 +111,68 @@ export class Ed25519BIP32 {
     }
     const [pk, sk, prefix, chaincode] = indices.reduce(
       deriveIndex,
-      deriveIndex([keychain.pk, keychain.sk, keychain.prefix, keychain.chaincode], indices.shift())
+      deriveIndex([keychain.pk, keychain.sk, keychain.prefix!, keychain.chaincode], indices.shift())
     );
     return { pk, sk, prefix, chaincode };
   }
 }
+
+function deriveSecp256k1Helper(index: number | undefined = 0, chaincode: bigint, pk: bigint, sk?: bigint): Buffer {
+  const data = Buffer.alloc(33 + 4);
+  if (((index >>> 0) & 0x80000000) === 0) {
+    bigIntToBufferBE(pk, 33).copy(data);
+  } else {
+    if (sk === undefined) {
+      throw new Error("Can't performed hardened derivation without private key");
+    }
+    data[0] = 0;
+    bigIntToBufferBE(sk, 32).copy(data, 1);
+  }
+  data.writeUInt32BE(index, 33);
+  return createHmac('sha512', bigIntToBufferBE(chaincode, 32)).update(data).digest();
+}
+
+export class BIP32 {
+  static curve: Secp256k1Curve = new Secp256k1Curve();
+
+  publicDerive(keychain: PublicKeychain, path: string): PublicKeychain {
+    const indices = path
+      .replace(/^m?\//, '')
+      .split('/')
+      .map((index) => parseInt(index, 10));
+    function deriveIndex(acc: bigint[], index: number | undefined): bigint[] {
```

Comment:
NIT: The derive index method also only have small differences, can we make it common and re-use to reduce duplicated code?

Reply:
I don't think deriveIndex is reusable between Ed25519 and secp256k1, at least not to a degree that code is simplified.

Reply:
Was talking about within secp256k1 public and privateDerive functions, but its okay